libdnf/system/state.cpp

```diff
@@ -222,8 +235,30 @@ void State::set_package_reason(const std::string & na, transaction::TransactionI
         reason_str);
 
     package_states[na].reason = reason_str;
+
+    // invalidate the package by reasons cache
```

Comment:
I think the set operation invalidate the cache of package reasons. But the comment is somehow cryptic for me.